src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import mate.jdbc.exeptions.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufactureRequest = "INSERT INTO manufacturers(name, country) values(?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufactureStatement =
+                        connection.prepareStatement(insertManufactureRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createManufactureStatement.setString(1, manufacturer.getName());
+            createManufactureStatement.setString(2, manufacturer.getCountry());
+            createManufactureStatement.executeUpdate();
+            ResultSet generatedKeys = createManufactureStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                manufacturer.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant create new data in database", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getAllManufactureRequest = "SELECT * FROM manufacturers WHERE is_deleted = 0";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturesStatement = connection
+                        .prepareStatement(getAllManufactureRequest);) {
+            ResultSet resultSet = getAllManufacturesStatement
+                                    .executeQuery(getAllManufactureRequest);
+            while (resultSet.next()) {
+                if (Objects.equals(resultSet.getObject("id", Long.class), id)) {
+                    return Optional.of(createManufacturer(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant get data by id " + id + "from database", e);
+        }
+        return Optional.of(new Manufacturer());
```

Comment:
It's better not to do that. Potentially, you can’t track in any way whether you received the necessary object or not. It is not null and optional is not empty, so we will safely access the fields, and we will get the wrong result and errors.
```suggestion
        return Optional.empty();
```